classes/permission.php

```diff
@@ -42,4 +42,42 @@ public static function can_vote(\context_module $context): bool {
     public static function require_can_vote(\context_module $context) {
         require_capability('mod/sortvoting:vote', $context);
     }
+
+
+    /**
+     * User can see results.
+     *
+     * @param \stdClass $sortvoting
+     * @param \context_module $context
+     * @return bool
+     */
+    public static function can_see_results(\stdClass $sortvoting, \context_module $context): bool {
+        global $DB, $USER;
+
+        $existingvotes = $DB->get_records_menu(
+            'sortvoting_answers',
+            [
+                'sortvotingid' => $sortvoting->id,
+                'userid' => $USER->id,
+            ],
+            'id ASC',
+            'optionid, position'
+        );
+        $studenthasvoted = !empty($existingvotes);
```

Comment:
This block (from L57) can be replaced with a single `$DB->record_exists(...)` call, no need to create a list and count it